internal/services/contracts_events_consumer.go

```diff
@@ -102,34 +102,24 @@ func NewContractsEventsConsumer(pdb db.Store, log *zerolog.Logger, settings *con
 	}
 }
 
-func (c *ContractsEventsConsumer) ProcessContractsEventsMessages(messages <-chan *message.Message) {
-	for msg := range messages {
-		err := c.processMessage(msg)
-		if err != nil {
-			c.log.Err(err).Msg("error processing contract events messages")
-		}
-	}
-}
-
-func (c *ContractsEventsConsumer) processMessage(msg *message.Message) error {
-	// Keep the pipeline moving no matter what.
-	defer func() { msg.Ack() }()
+func (c *ContractsEventsConsumer) RunConsumer() error {
+	ctx := context.Background()
 
-	// Deletion messages. We're the only actor that produces these, so ignore them.
-	if msg.Payload == nil {
-		return nil
+	if err := kafka.Consume[*shared.CloudEvent[json.RawMessage]](ctx, kafka.Config{
```

Comment:
Can we just return this error value to the caller? We're calling `.Fatal` here and in the caller, it seems duplicative.

Reply:
Do we need to provide this type parameter, by the way? I feel like Go can infer this.